### PR TITLE
new: Add logging for `linode_image` and `linode_images` resources/data sources

### PR DESCRIPTION
**Pull request by ykim-1**
## 📝 Description

Add logging for following resource and data source:
- linode_image (resource and data source)
- linode_images (data source)

## ✔️ How to Test

1. Enable trace logging for the Linode provider:
```
export TF_LOG=TRACE
export TF_LOG_PROVIDER_LINODE=TRACE
```
2. Inside of a Terraform provider sandbox environment (e.g. dx-devenv), apply the following configuration:
```
# ...
terraform {
  required_providers {
    // Running sandbox targets will automatically
    // use the local provider in repos/terraform-provider-linode
    linode = {
      source  = "linode/linode"
      version = "2.6.0"
    }
  }
}

resource "linode_instance" "foo" {
  type = "g6-nanode-1"
  region = "us-central"
  image = "linode/ubuntu22.04"
  root_pass = "insecure-p4ssw0rd!!"
}

resource "linode_image" "bar" {
  label = "foo-sda-image"
  description = "Image taken from foo"
  disk_id = linode_instance.foo.disk.0.id
  linode_id = linode_instance.foo.id
}

resource "linode_instance" "bar_based" {
  type = linode_instance.foo.type
  region = "eu-west"
  image = linode_image.bar.id
}

data "linode_images" "specific-images" {
  filter {
    name = "label"
    values = ["Debian 11"]
  }

  filter {
    name = "is_public"
    values = ["true"]
  }
}

output "image_id" {
  value = data.linode_images.specific-images.images.0.id
}

data "linode_images" "all-images" {}

output "image_ids" {
  value = data.linode_images.all-images.images.*.id
}
```

## 📷 Preview

**If applicable, include a screenshot or code snippet of this change. Otherwise, please remove this section.**

### CODE REVIEW

File: linode/image/resource.go [hunk 1]
```diff
@@ -86,6 +92,9 @@ func createResource(ctx context.Context, d *schema.ResourceData, meta interface{
 func createResourceFromLinode(
 	ctx context.Context, d *schema.ResourceData, meta interface{},
 ) diag.Diagnostics {
+	ctx = populateLogAttributes(ctx, d)
```
Comment (lgarber-akamai) [inline linode/image/resource.go:95]: I think this can be dropped since no image will exist at this point 👍

File: linode/image/resource.go [hunk 2]
```diff
@@ -135,6 +153,9 @@ func createResourceFromLinode(
 func createResourceFromUpload(
 	ctx context.Context, d *schema.ResourceData, meta interface{},
 ) diag.Diagnostics {
```
Comment (lgarber-akamai) [inline linode/image/resource.go:155]: Same as above

File: linode/image/resource.go [hunk 3]
```diff
@@ -135,6 +153,9 @@ func createResourceFromLinode(
...
+	ctx = populateLogAttributes(ctx, d)
```
Comment (lgarber-akamai) [inline linode/image/resource.go:156]: Same as above

**Review by lgarber-akamai — APPROVED** [review]
LGTM, just a few tiny adjustments 🙂

**Review by zliang-akamai — APPROVED** [review]
Agree to @lgarber-akamai's comment. Everything else looks great!

**Pull request merged**